brainscore_vision/model_helpers/activations/core.py

```diff
@@ -125,31 +177,99 @@ def register_stimulus_set_hook(self, hook):
+            # add this batch to layer_activations
+            for layer_name, layer_output in batch_activations.items():
+                layer_activations.setdefault(layer_name, []).append(layer_output)
 
-        return layer_activations
+        # fast concat all batches
+        for layer_name, layer_outputs in layer_activations.items():
+            layer_activations[layer_name] = np.concatenate(layer_outputs)
 
-    def _get_batch_activations(self, inputs, layer_names, batch_size):
+        return layer_activations  # this is all batches
+
+    def _get_batch_activations(self, inputs, layer_names, batch_size, require_variance: bool = False,
+                               trial_number: int = 1):
         inputs, num_padding = self._pad(inputs, batch_size)
         preprocessed_inputs = self.preprocess(inputs)
+        preprocessed_inputs = self.translate_images(preprocessed_inputs, inputs, trial_number, require_variance)
         activations = self.get_activations(preprocessed_inputs, layer_names)
         assert isinstance(activations, OrderedDict)
         activations = self._unpad(activations, num_padding)
+        if require_variance:
+            self.remove_temporary_files(preprocessed_inputs)
         return activations
 
+    def set_visual_degrees(self, visual_degrees: float) -> None:
+        """
+        A method used by ModelCommitments to give the ActivationsExtractorHelper their visual degrees for
+        performing microsaccades.
+        """
+        self._visual_degrees = visual_degrees
+
+    def translate_images(self, images: List[Union[str, np.ndarray]], image_paths: List[str], trial_number: int,
+                         require_variance: bool) -> List[str]:
+        """A method that translates images according to selected microsaccades, if microsaccades are required."""
+        output_images = []
+        for index, image_path in enumerate(image_paths):
+            # When microsaccades are not used, skip computing them and return the base images.
+            #  This iteration could be entirely skipped, but recording microsaccades for all images regardless
+            #  of whether variance is required or not is convenient for adding an extra presentation dimension
+            #  in the layer assembly later to avoid collapse, or otherwise extraneous mock dims.
+            #  The method could further be streamlined by calling `self.get_image_with_shape()` and
+            #  `self.select_microsaccade` for all images regardless of require_variance, but it seems like a bad
+            #  idea to introduce cv2 image loading for all models and images, regardless of whether they are actually
+            #  microsaccading.
+            if not require_variance:
+                self.microsaccades['pixels'][image_path] = [(0., 0.)]
+                self.microsaccades['degrees'][image_path] = [(0., 0.)]
+                output_images.append(images[index])
```

Comment:
I'm confused about the difference between `images` and `image_paths`. I thought we could just `output_images.append(image_path)` but I'm not sure how the `images` are different

Reply:
for non-tf models `images` at this point are already tensors. added a comment